featuremanagement/azuremonitor/_send_telemetry.py

```diff
@@ -53,50 +55,53 @@ def publish_telemetry(evaluation_event: EvaluationEvent) -> None:
+
+    feature = evaluation_event.feature
+
+    if not feature:
         return
-    event[FEATURE_NAME] = evaluation_event.feature.name
-    event[ENABLED] = str(evaluation_event.enabled)
-    event["Version"] = EVALUATION_EVENT_VERSION
+
+    event: Dict[str, Optional[str]] = {
+        FEATURE_NAME: feature.name,
+        ENABLED: str(evaluation_event.enabled),
+        "Version": EVALUATION_EVENT_VERSION,
+    }
+
+    reason = evaluation_event.reason
+    variant = evaluation_event.variant
 
     # VariantAllocationPercentage
-    if evaluation_event.reason and evaluation_event.reason != VariantAssignmentReason.NONE:
-        if evaluation_event.variant:
-            event[VARIANT] = evaluation_event.variant.name
-        event[REASON] = evaluation_event.reason.value
-
-        if evaluation_event.reason == VariantAssignmentReason.DEFAULT_WHEN_ENABLED:
-            allocation_percentage = 0
-
-            if evaluation_event.feature.allocation and evaluation_event.feature.allocation.percentile:
-                for allocation in evaluation_event.feature.allocation.percentile:
-                    if (
-                        evaluation_event.variant
-                        and allocation.variant == evaluation_event.variant.name
-                        and allocation.percentile_to
-                    ):
-                        allocation_percentage += allocation.percentile_to - allocation.percentile_from
+    if reason and reason != VariantAssignmentReason.NONE:
+        if variant:
+            event[VARIANT] = variant.name
+        event[REASON] = reason.value
+
+        allocation_percentage = 0
+        if (
+            reason == VariantAssignmentReason.DEFAULT_WHEN_ENABLED
+            and feature.allocation
+            and feature.allocation.percentile
+        ):
+            for allocation in feature.allocation.percentile:
+                if variant and allocation.variant != variant.name and allocation.percentile_to:
+                    allocation_percentage += allocation.percentile_to - allocation.percentile_from
 
             event["VariantAssignmentPercentage"] = str(100 - allocation_percentage)
-        elif evaluation_event.reason == VariantAssignmentReason.PERCENTILE:
-            if evaluation_event.feature.allocation and evaluation_event.feature.allocation.percentile:
-                allocation_percentage = 0
-                for allocation in evaluation_event.feature.allocation.percentile:
-                    if (
-                        evaluation_event.variant
-                        and allocation.variant == evaluation_event.variant.name
-                        and allocation.percentile_to
-                    ):
+        elif reason == VariantAssignmentReason.PERCENTILE:
+            if feature.allocation and feature.allocation.percentile:
+
+                for allocation in feature.allocation.percentile:
+                    if variant and allocation.variant == variant.name and allocation.percentile_to:
```

Comment:
The percentile_to check doesn't make sense to me either. Unless it's checking for None. In which case we should check percentile_from as well.

Reply:
`percentile_from` can't be `None` but `percentile_to` can. To is type `Optional[int]` from is just in `int`.

Reply:
Can you please review just the last commit https://github.com/microsoft/FeatureManagement-Python/pull/45/commits/0655a7211c9c0855ff52494698dd19909c0016ed. This is the change that would need to be made to remove the optional.

Reply:
> percentile_from can't be None but percentile_to can

Interesting. percentile_to should not be Optional[int]. I don't see any reason it could be optional.

Reply:
> Can you please review just the last commit https://github.com/microsoft/FeatureManagement-Python/commit/0655a7211c9c0855ff52494698dd19909c0016ed. This is the change that would need to be made to remove the optional.

LGTM

Reply:
> > percentile_from can't be None but percentile_to can
> 
> Interesting. percentile_to should not be Optional[int]. I don't see any reason it could be optional.

The reason it was, was that I didn't like defaulting to have it being 100, so I put it as optional with a `None` default value. The commit above changed it. We should be good now.

Reply:
Could we make the default 0? Might satisfy both concerns